hash/hash.go

```diff
@@ -0,0 +1,101 @@
+// Package hash provides utility functions to process complex data types, like
+// data streams, files, or sequences of structs of different types.
+package hash
+
+import (
+	"bytes"
+	"errors"
+	h "hash"
+	"io"
+	"os"
+
+	"encoding"
+	"reflect"
+)
+
+var defaultChunkSize = 1024
+
+// Stream returns the hash of a data stream separated into chunks of the defaultChunkSize.
+func Stream(hash h.Hash, stream io.Reader) ([]byte, error) {
+	return StreamChunk(hash, stream, defaultChunkSize)
+}
+
+// StreamChunk returns the hash of a data stream separated into chunks of the given byte size.
+func StreamChunk(hash h.Hash, stream io.Reader, size int) ([]byte, error) {
+	if size < 1 {
```

Comment:
can't we use `io.Copy` here? Why do we need to work with chunks? IIRC, this was my code, and I take the blame ;) Now I'd simply use `io.Copy`.